### ISSUE TITLE
2D weights for multitask models

### ISSUE THREAD

**Issue by peastman** [keras-team/keras#9323]
Keras requires `sample_weight` to be a 1D array with a single number for each sample.  That isn't sufficient for multitask models.  In that case, the model predicts multiple values for each sample and the weights need to be a 2D matrix of shape `(n_samples, n_tasks)`.  The outputs and labels will be two or more dimensional of shape `(n_samples, n_tasks, ...)`, depending on the problem.  For a simple regression problem, they might just be 2D.  For a classification problem, they would probably have shape `(n_samples, n_tasks, n_classes)`.  For a computer vision model, they might have shape `(n_samples, n_tasks, width, height)`.  And so on.  In any case, the loss function needs to be computed for each sample/task pair.  That produces a 2D matrix of losses that needs to be multiplied by the matrix of weights.

I'd like to suggest adding a `sample_weight_mode='multitask'` that would implement this case.  Currently, implementing models of this sort in Keras just seems to be impossible.

This is closely related to #2408 and #2594.  Those both got closed without any apparent resolution, so that's why I'm opening a new feature request.

**Comment by hobson** [comment]
Any progress on this? What is the correct way to train a multitask model on data that may contain missing labels for either or both tasks, and you want to deweight (ignore) a sample for one task or the other, not both?

**Comment by peastman** [comment]
Any explanation for why this was closed?  Is there a better way of achieving the same goal?  If so, what is it?

This appears to now be the third request for this feature to have gotten closed without explanation.

**Comment by Ram-WD** [comment]
is there any news ?
